Framework/Core/src/DataProcessingHelpers.cxx

```diff
@@ -19,18 +19,18 @@
 
 namespace o2::framework
 {
-void DataProcessingHelpers::sendEndOfStream(FairMQDevice& device, OutputChannelSpec const& channel)
+void DataProcessingHelpers::sendEndOfStream(FairMQDevice& device, std::string const& channelName)
```

Comment:
Is it really required to change it to a string? I personally like the forward declaration better...

Reply:
Moreover (even if not critical in this case), using a proper type would allow us to cache the pointer to the channel (rather than doing the lookup by string).

Reply:
ok, I will check

Reply:
Getting back to that. I share your opinion to keep the tree of includes small. Since forward declaration is not possible for std::string we could either switch to `const char*` or introduce a specific class `ChannelName` holding the string and providing initialization and conversion function. But since we want to pass the argument by ref to allow the forward declaration, in the caller we would need to specify `ChannelName(...)`.

Reply:
Ah, maybe I do not need to change the signature of this helper at all. I'm getting the output routes from the device spec, but I can equally good use the output channels.
```
struct DeviceSpec {
  ...
  std::vector<OutputChannelSpec> outputChannels;
  ...
  std::vector<OutputRoute> outputs;
  std::vector<ForwardRoute> forwards;
  ...
}
```
Now I'm thinking its better anyhow to use the `outputChannels` vector and then I have `OutputChannelSpec` as argument.

Reply:
I have updated the PR to use `outputChannels` instead of `outputRoutes`